### PR TITLE
Tests for the Sample App

### PR DESCRIPTION
**Pull request by xdaDaveShaw**
As I mentioned in #19 I've added some tests to the existing sample app, and also restructured the folders so that they are a little less verbose, and now support >1 sample app.

As with [Suave](https://github.com/SuaveIO/suave/tree/master/src/Suave.Testing), I think there's merit in a testing library that can support the main library. I have some basic functions in the top of the test file, but there's not enough for even a basic alpha (I'm really struggling with Ionide at the moment so refactoring is hard without intellisense / code lens / basic IDE).

Also, I'm not sure sure I like using xUnit.net out of the box, I know there's Expecto, Unquote, FsUnit and probably more, but I didn't want to diverge too far from the main library.

If you think there's merit in either a separate testing lib and/or changing the test framework - I can create some new issues

### CODE REVIEW

File: samples/SampleApp/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Sample Application
+
+A sample application for AspNetCore.Lambda.
+
+Very small, only contains the basics:
+
+- Routing.
+- Cookie authentication.
+- Error handling.
+- Intergration Testing.
```
Comment (dustinmoris) [inline samples/SampleApp/README.md]: Typo - "Integration"

File: samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="1.1.*" />
+    <PackageReference Include="xunit" Version="2.2.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <DotNetCliToolReference Include="dotnet-compile-fsc">
+      <Version>1.0.0-preview2-020000</Version>
+    </DotNetCliToolReference>
+  </ItemGroup>
```
Comment (dustinmoris) [inline samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj]: `dotnet-compile-fsc` is not required anyore. The package `FSharp.NET.Sdk` includes the fsc now.

File: samples/SampleApp/README.md (hunk 1 above)
Comment (xdaDaveShaw) [inline samples/SampleApp/README.md]: I swear I fixed that once, maybe forgot to save it.
Pushed a correction.

File: samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj (hunk 2 above)
Comment (xdaDaveShaw) [inline samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj]: OK, Removed, will push any other comments in a single commit

**Comment by dustinmoris** [comment]
> I think there's merit in a testing library that can support the main library.

Agreed. In the beginning I would prefer to create a separate module inside the main library with all the test functions and only when it grows big enough move it out into a separate library.

> Also, I'm not sure sure I like using xUnit.net out of the box, I know there's Expecto, Unquote, FsUnit and probably more, but I didn't want to diverge too far from the main library.

I have used almost exclusively NUnit in the last years and used XUnit in this and in another project for the first time simply to try it out. I was always quite happy with NUnit in C# and NUnit + FsUnit in F#. Basically I don't mind using other test frameworks or additional helper libraries to make writing tests quicker and easier. If it would be a complete different framework then I'd be interested in the advantages/trade-offs over using xunit or nunit.

> I'm really struggling with Ionide at the moment so refactoring is hard without intellisense / code lens / basic IDE

Yeah the tooling for F# with .NET Core is not great yet. Ionide intellisense works pretty well, but you have to restore and build the project before you open it in VS Code. Give it another try and let me know if you have any questions. The maintainer of Ionide is pretty responsive as well when you tweet him or try to message him on Slack/Gitter.

File: samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj (hunk 2 above)
Comment (dustinmoris) [inline samples/SampleApp/SampleApp.Tests/SampleApp.Tests.fsproj]: That's it, looks good to me!

**Comment by dustinmoris** [comment]
Thank you! :)

**Pull request merged**